snorkel/models/candidate.py

```diff
@@ -30,8 +30,42 @@ def append(self, item):
     def remove(self, item):
         self.candidates.remove(item)
 
+    def difference(self, other, name, session):
```

Comment:
This `difference` function (along with the `intersection` function) can be removed. 

The functionality would be nice to have in the future, but for these functions to work they require the ability to add a Candidate to multiple CandidateSets (which isn't currently possible).